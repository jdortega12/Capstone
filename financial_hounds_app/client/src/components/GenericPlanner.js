import React from "react";
import "../styles/GenericPlanner.css";
import { useState } from 'react';

const GenericPlanner = () => {

  const [income, setIncome] = useState("");
  let [budget, setBudget] = useState({
    necesities: 0,
    wants: 0,
    savings: 0,
  });

  const computeBudget = () => {
    var strippedIncome = income.replace(/,/g,'');
    const userIncome = Number(strippedIncome);
    const calculatedNecessities = userIncome * 0.5;
    const calculatedWants = userIncome * 0.3;
    const calculatedSavings = userIncome * 0.2;

    setBudget({
      necesities: calculatedNecessities,
      wants: calculatedWants,
      savings: calculatedSavings,
    }); 
  };

    return (
      <div className="genericPlanner">
        <div>
        <h1 className="headerGenericPlanner"> Generic Planner </h1>
          <p className="paragraphGenericPlanner">
            The Generic Planner will break your monthly after-tax income apart using the '50/30/20' budgeting method.
            The method sets apart 50% of your monthly income for necessities, 30% for wants, and 20% for savings and debt payments.
          </p>
          <row>
            <label className="myLabelGenericPlanner">Income</label>
            <input
              name="income"
              type="text" 
              value = {income}
              onChange = {(e) => setIncome(e.target.value)}
              className="myInputGenericPlanner" 
              placeholder="Enter monthly income" />
            <button onClick={()=> computeBudget()} className="myButtonGenericPlanner">
              Compute Budget
            </button>
          </row>
            <div className="myBoxGenericPlanner">
              <h1 className="myPanelGenericPlanner">Necessities: ${budget.necesities}</h1>
              <h1 className="myPanelGenericPlanner">Wants: ${budget.wants}</h1>
              <h1 className="myPanelGenericPlanner">Savings: ${budget.savings}</h1>
            </div>
        </div>
        <div>

        </div>
      </div>
    );
  }
  
export default GenericPlanner;