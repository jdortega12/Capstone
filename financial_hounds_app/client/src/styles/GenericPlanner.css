.genericPlanner {
    width: 100%;
    height: 100vh;
    display: flex;
    background-color: #0d4b34;
    font-family: "Gill Sans", "Gill Sans MT", Calibri, "Trebuchet MS", sans-serif;
  }

.headerGenericPlanner {
    font-size: 75px;
    font-weight: 50;
    color: white;
    margin-left: 25px;
    text-align: center;
  }

.paragraphGenericPlanner {
    margin-left: 25px;
    margin-left: 33.3%;
    text-align: center;
    border: 3px solid;
    border-color: white;
    font-size: 20px;
    font-weight: lighter;
    color: white;
    width: 37%;
    margin-bottom: 16px;
  }

.myLabelGenericPlanner{
    color: white;
    margin-right: 100%;
    margin-left: 42%;
    font-size: 20px;
}

.myInputGenericPlanner{
    width: 20%;
    margin-left: 42%;
    font-size: 20px;
}

.myButtonGenericPlanner{
  margin-left: 43%;
  margin-top: 30px;
  border: none;
  border-radius: 5px;
  cursor: pointer;
  background-color: #282c34;
  height: 10%;
  width: 18%;
  color: white;
  text-decoration: none;
  font-size: 20px;
  padding: 20px;
}

.myBoxGenericPlanner{
  padding: 20px;
  display: flex;
}

.myPanelGenericPlanner{
  flex: 1;
  color: white;
  text-align: center;
  border: 3px solid;
  border-color: white;
}

/*
    margin-right: 100%;
    margin-left: 50px;

    margin-bottom: 5%;
    margin-left: 50px;

*/



