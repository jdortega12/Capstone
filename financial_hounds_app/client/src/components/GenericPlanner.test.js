import React from 'react';
import GenericPlanner from '../components/GenericPlanner';
import renderer from 'react-test-renderer';

test("Generic Planner Snapshot Test", ()=> {
    const component = renderer.create(
        <GenericPlanner />
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});