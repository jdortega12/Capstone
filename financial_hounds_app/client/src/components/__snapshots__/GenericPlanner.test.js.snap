// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Generic Planner Snapshot Test 1`] = `
<div
  className="genericPlanner"
>
  <div>
    <h1
      className="headerGenericPlanner"
    >
       Generic Planner 
    </h1>
    <p
      className="paragraphGenericPlanner"
    >
      The Generic Planner will break your monthly after-tax income apart using the '50/30/20' budgeting method. The method sets apart 50% of your monthly income for necessities, 30% for wants, and 20% for savings and debt payments.
    </p>
    <row>
      <label
        className="myLabelGenericPlanner"
      >
        Income
      </label>
      <input
        className="myInputGenericPlanner"
        name="income"
        onChange={[Function]}
        placeholder="Enter monthly income"
        type="text"
        value=""
      />
      <button
        className="myButtonGenericPlanner"
        onClick={[Function]}
      >
        Compute Budget
      </button>
    </row>
    <div
      className="myBoxGenericPlanner"
    >
      <h1
        className="myPanelGenericPlanner"
      >
        Necessities: $
        0
      </h1>
      <h1
        className="myPanelGenericPlanner"
      >
        Wants: $
        0
      </h1>
      <h1
        className="myPanelGenericPlanner"
      >
        Savings: $
        0
      </h1>
    </div>
  </div>
  <div />
</div>
`;
